### PR TITLE
Jetpack: only display Backup and Scan if available

### PR DESCRIPTION
**Pull request by leandroalonso**
Part of #15193

WPKit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/326

This PR displays the "Backup" and "Scan" options under My Site *only* if available.

#### When available

1. Open the app
2. Swipe to refresh in the blog list or just wait a few seconds
3. Tap a blog that has Backup and/or Scan available
4. Make sure the option is shown

#### When not available

1. Open the app
2. Swipe to refresh in the blog list or just wait a few seconds
3. Tap a blog that *does not* has Backup and/or Scan available
4. Make sure the option is *not* shown

#### Login

1. Do a fresh install
2. Login
3. Check that Backup and/or Scan is correctly displayed the first time you see the blog detail

@emilylaguna can you please let me know if there's more code to delete from the previous implementation?

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/WordPressTest/JetpackCapabilitiesServiceTests.swift [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import Foundation
+import XCTest
+
+@testable import WordPress
+
+/// GravatarService Unit Tests
```
Comment (momo-ozawa) [inline WordPress/WordPressTest/JetpackCapabilitiesServiceTests.swift]: Old comment? 👀

**Review by momo-ozawa — COMMENTED** [review]
### Slack discussion notes:
- Backup/Scan isn’t showing up for the digital ocean Jetpack test sites
- This is due to backup/scan capabilities not being in included in the response

File: WordPress/WordPressTest/JetpackCapabilitiesServiceTests.swift (hunk 1 above)
Comment (leandroalonso) [reply WordPress/WordPressTest/JetpackCapabilitiesServiceTests.swift]: Uh oh 0fbe36c8b15b16f2973f16ada601d37ce1996bb2

**Comment by leandroalonso** [comment]
@momo-ozawa I changed the implementation and it should work now with all sites. :)

File: WordPress/Classes/Services/BlogService.m [hunk 2]
```diff
@@ -941,19 +943,21 @@ - (BlogDetailsHandler)blogDetailsHandlerWithBlogObjectID:(NSManagedObjectID *)bl
                                        completionHandler:(void (^)(void))completion
 {
     return ^void(RemoteBlog *remoteBlog) {
-        [self.managedObjectContext performBlock:^{
-            NSError *error = nil;
-            Blog *blog = (Blog *)[self.managedObjectContext existingObjectWithID:blogObjectID
-                                                                           error:&error];
-            if (blog) {
-                [self updateBlog:blog withRemoteBlog:remoteBlog];
+        [[[JetpackCapabilitiesService alloc] init] syncWithBlogs:[NSArray arrayWithObject:remoteBlog] success:^(NSArray<RemoteBlog *> *blogs) {
```
Comment (emilylaguna) [inline WordPress/Classes/Services/BlogService.m]: You can use the `@[]` shorthand to create an array

```suggestion
        [[[JetpackCapabilitiesService alloc] init] syncWithBlogs:@[remoteBlog] success:^(NSArray<RemoteBlog *> *blogs) {
```

File: WordPress/Classes/Services/JetpackCapabilitiesService.swift [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import WordPressKit
+
+@objc class JetpackCapabilitiesService: NSObject {
+
+    let capabilitiesServiceRemote: JetpackCapabilitiesServiceRemote
+
+    init(capabilitiesServiceRemote: JetpackCapabilitiesServiceRemote? = nil) {
+        let api = WordPressComRestApi.defaultApi(in: ContextManager.shared.mainContext, localeKey: WordPressComRestApi.LocaleKeyV2)
```
Comment (emilylaguna) [inline WordPress/Classes/Services/JetpackCapabilitiesService.swift]: What do you think about injecting the context in `init`?

**Review by emilylaguna — CHANGES_REQUESTED** [review]
Working well! Left a couple comments below.

File: WordPress/Classes/Services/BlogService.m (hunk 2 above)
Comment (leandroalonso) [reply WordPress/Classes/Services/BlogService.m]: 👀 

0918438ae8333ba13cf325ab12648e762b7bb81c

File: WordPress/Classes/Services/JetpackCapabilitiesService.swift (hunk 3 above)
Comment (leandroalonso) [reply WordPress/Classes/Services/JetpackCapabilitiesService.swift]: 24226bef15e87767975b6a592cdf1037f3de51b8

**Review by emilylaguna — APPROVED** [review]
LGTM! 🚢

**Pull request merged**